trieste/bayesian_optimizer.py

```diff
@@ -1092,6 +1099,8 @@ def early_stop_callback(
         _models: Mapping[Tag, TrainableProbabilisticModel],
         _acquisition_state: object,
     ) -> bool:
+        if minimum_step_number is not None and logging.get_step_number() < minimum_step_number:
```

Comment:
is perhaps getting the step variable this way problematic? recall the discussion with Alan about this being a global variable and when we have parallel processes it could give incorrect values?

we should probably have step variable as an attribute of the `BayesianOptimizer` object and pass it to the function? or can it be inferred from the state?

Reply:
We can address this if and when we change how step_number is exposed, though I suspect we will find a way to make it accessible in the separate processes without breaking the current API. Certainly we don't want to change the EarlyStopCallback function type without a strong reason.